client/src/components/ContentNavigator/ContentDataProvider.ts

```diff
@@ -214,6 +217,11 @@ class ContentDataProvider
     const isContainer = getIsContainer(item);
     const uri = await this.model.getUri(item, false);
 
+    // Cache the URI to parent mapping
+    if (item.parentFolderUri) {
+      this.uriToParentMap.set(item.uri, item.parentFolderUri);
+    }
```

Comment:
Top level folder doesn't have a parentFolder.
Currently in `isDescendantOf`, as long as there's not a parentFolder, it will call `this.model.getResourceByUri`. And again the parentFolder is still empty. They're unnecessary network calls recursively until hit the 10 times limit. They should be avoided.
The fallback call is only used in case an item didn't go through this line. But if there's not a parentFolderUri here, it means this item really doesn't have a parent. Calling `this.model.getResourceByUri` is unnecessary and will result the same.
Perhaps something like

```suggestion
   const STOP_SIGN = '__STOP_SIGN__';
   this.uriToParentMap.set(item.uri, item.parentFolderUri ?? STOP_SIGN);
```